lib/rubocop/cop/rspec/rails/http_status.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Rails
+        # Enforces use of symbolic or numeric value to describe HTTP status.
+        #
+        # @example `EnforcedStyle: symbol` (default)
+        #   # bad
+        #   it { is_expected.to have_http_status 200 }
+        #   it { is_expected.to have_http_status 404 }
+        #
+        #   # good
+        #   it { is_expected.to have_http_status :ok }
+        #   it { is_expected.to have_http_status :not_found }
+        #   it { is_expected.to have_http_status :success }
+        #   it { is_expected.to have_http_status :error }
+        #
+        # @example `EnforcedStyle: numeric`
+        #   # bad
+        #   it { is_expected.to have_http_status :ok }
+        #   it { is_expected.to have_http_status :not_found }
+        #
+        #   # good
+        #   it { is_expected.to have_http_status 200 }
+        #   it { is_expected.to have_http_status 404 }
+        #   it { is_expected.to have_http_status :success }
+        #   it { is_expected.to have_http_status :error }
+        #
+        class HttpStatus < Cop
+          begin
+            require 'rack/utils'
+            AUTOCORRECTS = true
+          rescue LoadError
+            AUTOCORRECTS = false
+          end
+
+          include ConfigurableEnforcedStyle
+
+          MSG = 'Prefer `%<prefer>s` over `%<current>s` '\
+                'to describe HTTP status code.'.freeze
+          WHITELIST_STATUS = %i[error success missing redirect].freeze
+
+          def_node_matcher :http_status, <<-PATTERN
+            (send nil? :have_http_status ${int sym})
+          PATTERN
+
+          def on_send(node) # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
I haven't been able to avoid the `CyclomaticComplexity` offense without disabling the cop. Do you have any idea how I could improve my code ?

Reply:
I’m not sure how to fix it. Will take a closer look later.